flepimop/gempyor_pkg/src/gempyor/batch.py

```diff
@@ -1,23 +1,365 @@
 """
-Generate manifest files containing batch run metadata.
+Functionality for creating and submitting batch jobs.
 
-This module provides functionality for working with and executing batch jobs, either on
-HPC or AWS environments.
-
-Functions:
-    write_manifest: A utility to generate a manifest json file with metadata for a batch
-        job.
+This module provides functionality for required for batch jobs, including creating 
+metadata and job size calculations for example.
 """
 
-__all__ = ["write_manifest"]
+__all__ = ["BatchSystem", "JobSize", "JobResources", "JobTimeLimit", "write_manifest"]
 
 
+from dataclasses import dataclass
+from datetime import timedelta
+from enum import Enum, auto
 import json
+import math
 from pathlib import Path
 import sys
-from typing import Any
+from typing import Any, Literal
+
+from .utils import _git_head
+
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    Self = Any
+
+
+class BatchSystem(Enum):
+    """
+    Enum representing the various batch systems that flepiMoP can run on.
+    """
+
+    AWS = auto()
+    LOCAL = auto()
+    SLURM = auto()
+
+    @classmethod
+    def from_options(
+        cls,
+        batch_system: Literal["aws", "local", "slurm"] | None,
+        aws: bool,
+        local: bool,
+        slurm: bool,
```

Comment:
why these flags? they are mutually exclusive, and the selection is captured by `batch_system`

Reply:
Discussed offline, but adding note here for other reviewers.

https://github.com/HopkinsIDD/flepiMoP/blob/ad11faea053045f6e61fab546ef9335b518ff726/flepimop/gempyor_pkg/src/gempyor/batch.py#L878-L904

The reason for this was to resolve these options from the prior pull request, GH-394, into a single consistent answer. Have removed for now and will rethink how to implement in a follow up PR to accommodate the dynamic batch systems added in 78d960a.